server.go

```diff
@@ -456,9 +457,18 @@ func (s *Server) startActorC(c context.Context, start *ActorStart) error {
 	// and capture panics that the actor raises.
 	go func() {
 		defer func() {
-			timeout, cancel := context.WithTimeout(context.Background(), s.cfg.Timeout)
-			s.registry.Deregister(timeout, nsName)
-			cancel()
+			var err error
+			retry.X(3, 3*time.Second,
+				func() {
+					timeout, cancel := context.WithTimeout(context.Background(), s.cfg.Timeout)
+					err = s.registry.Deregister(timeout, nsName)
+					cancel()
+					return err == nil
```

Comment:
We want to retry on failures, not successes, right?

```suggestion
					return err != nil
```

Reply:
ops, I also missed that reviewing the PR :)